electroncash/transaction.py

```diff
@@ -97,6 +97,24 @@ def parse_scriptSig(d, _bytes):
         print_error("cannot find address in input script", bh2u(_bytes))
         return
 
+    # p2sh, non m of n
+    multisig_match = [ opcodes.OP_0 ] + [ opcodes.OP_PUSHDATA4 ] * (len(decoded) - 1)
+    if  len(decoded) >= 3 and match_decoded(decoded, multisig_match):
+        pass
+    else:
+        redeem_script = None
```

Comment:
From skimming this -- it looks to me like this is just going to false-positive anything that has a push at the end and doesn't start with OP_0 and has less than 3 pushes. Including p2pk, p2kh, etc.